projects/packages/videopress/src/class-access-control.php

```diff
@@ -0,0 +1,399 @@
+		 *
+		 * @return array
+		 */
+		return (array) apply_filters( 'videopress_video_restriction_details', $video_restriction_details, $guid, $embedded_post_id, $selected_plan_id );
+	}
+
+	/**
+	 * Returns the default restriction_details for a video.
+	 *
+	 * @param bool $default_can_access The default auth.
+	 *
+	 * @return array
+	 **/
+	private function default_video_restriction_details( $default_can_access = false ) {
+		$restriction_details = array(
+			'version'              => '1',
+			'provider'             => 'auth',
+			'title'                => __( 'Unauthorized', 'jetpack-videopress-pkg' ),
+			'unauthorized_message' => __( 'Unauthorized', 'jetpack-videopress-pkg' ),
+			'can_access'           => $default_can_access,
+		);
+
+		return $restriction_details;
+	}
+
+	/**
+	 * Determines if the current user can view the provided video. Only ever gets fired if site-wide private videos are enabled.
+	 *
+	 * Filterable for 3rd party plugins.
+	 *
+	 * @param string $guid             The video id being checked.
+	 * @param int    $embedded_post_id The post id the video is embedded in or 0.
+	 * @param int    $selected_plan_id The plan id the earn block this video is embedded in has.
+	 */
+	public function is_current_user_authed_for_video( $guid, $embedded_post_id, $selected_plan_id = 0 ) {
+		if ( current_user_can( 'upload_files' ) ) {
+			return $this->filter_is_current_user_authed_for_video( true, $guid, $embedded_post_id );
+		}
+
+		$attachment = false;
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$video_info = video_get_info_by_guid( $guid );
+			if ( ! empty( $video_info ) ) {
+				$attachment = get_blog_post( $video_info->blog_id, $video_info->post_id );
+			}
+		} else {
+			$attachment = videopress_get_post_by_guid( $guid );
+		}
+
+		if ( ! $attachment ) {
+			return false;
+		}
+
+		$video_info = video_get_info_by_blogpostid( get_current_blog_id(), $attachment->ID );
+		if ( null === $video_info->guid ) {
+			return false;
+		}
+
+		$is_user_authed = false;
+
+		// Determine if video is public, private or use site default.
+		switch ( $video_info->privacy_setting ) {
+			case VIDEOPRESS_PRIVACY::IS_PUBLIC:
+				$is_user_authed = true;
+				break;
+			case VIDEOPRESS_PRIVACY::IS_PRIVATE:
+				$restriction_details = $this->build_restriction_details( $guid, $embedded_post_id, $selected_plan_id );
+				$is_user_authed      = $restriction_details['can_access'];
+				break;
+			case VIDEOPRESS_PRIVACY::SITE_DEFAULT:
+			default:
+				$is_videopress_private_for_site = Data::get_videopress_videos_private_for_site();
+				$is_user_authed                 = true;
+				if ( $is_videopress_private_for_site ) {
+					$restriction_details = $this->build_restriction_details( $guid, $embedded_post_id, $selected_plan_id );
+					$is_user_authed      = $restriction_details['can_access'];
+				}
+				break;
+		}
+
+		$this->update_video_restriction_details_on_wpcom( $guid, $restriction_details );
```

Comment:
I am struggling to understand why video  restriction details get updated on WPCOM every time `is_current_user_authed_for_video` is called...
At least shouldn't there be some cache (at least for the current request) to prevent multiple calls?

Reply:
@millerf I thought about doing it on save but decided against it. Since the privacy setting is impacted by parent blocks, filters etc, it seems like the most "late-bound" place to do it. 

I will go ahead and remove the call entirely for this version of the feature. The impact if removed will be that subscriber-only videos will display a generic "video is private" error when not viewed within the site.